File: api/src/main/java/tutmirleid/api/service/ScenarioContextOptionService.java
package tutmirleid.api.service;

import org.springframework.stereotype.Service;
import tutmirleid.api.entity.ScenarioContextOptionEntity;
import tutmirleid.api.repository.ScenarioContextOptionRepository;

import java.util.List;

@Service
public class ScenarioContextOptionService {
    private final ScenarioContextOptionRepository scenarioContextOptionRepository;

    public ScenarioContextOptionService(ScenarioContextOptionRepository scenarioContextOptionRepository) {
        this.scenarioContextOptionRepository = scenarioContextOptionRepository;
    }

    public List<ScenarioContextOptionEntity> getOptions() {
        return scenarioContextOptionRepository.findAll();
    }
}

File: api/src/scenario/contextOption/scenario.context.option.service.ts
import {HttpException, HttpStatus, Injectable} from "@nestjs/common";
import {InjectRepository} from "@nestjs/typeorm";
import {ScenarioContextOptionEntity} from "../../entity/scenario/scenario.context.option.entity";
import {Repository} from "typeorm";

@Injectable()
export class ScenarioContextOptionService {
    constructor(@InjectRepository(ScenarioContextOptionEntity) private scenarioContextOptionRepository: Repository<ScenarioContextOptionEntity>) {}

    async getOptions() {
        try {
            return await this.scenarioContextOptionRepository.find();
        } catch (error) {
            throw new HttpException(error, HttpStatus.INTERNAL_SERVER_ERROR)
        }
    }
}